src/native/external/zlib-ng.cmake

```diff
@@ -0,0 +1,161 @@
+    arch/arm/slide_hash_armv6.c
+    arch/arm/slide_hash_neon.c
+# ARM *.h
+    arch/arm/acle_intrins.h
+    arch/arm/arm_features.h
+    arch/arm/neon_intrins.h
+# Power *.c
+    arch/power/adler32_power8.c
+    arch/power/adler32_vmx.c
+    arch/power/chunkset_power8.c
+    arch/power/compare256_power9.c
+    arch/power/crc32_power8.c
+    arch/power/power_features.c
+    arch/power/slide_hash_power8.c
+    arch/power/slide_hash_vmx.c
+# Power *.h
+    arch/power/crc32_constants.h
+    arch/power/fallback_builtins.h
+    arch/power/power_features.h
+    arch/power/slide_ppc_tpl.h
+# RISCV *.c
+    arch/riscv/adler32_rvv.c
+    arch/riscv/chunkset_rvv.c
+    arch/riscv/compare256_rvv.c
+    arch/riscv/riscv_features.c
+    arch/riscv/slide_hash_rvv.c
+# RISCV *.h
+    arch/riscv/riscv_features.h
+# S390 *.c
+    arch/s390/crc32-vx.c
+    arch/s390/dfltcc_common.c
+    arch/s390/dfltcc_deflate.c
+    arch/s390/dfltcc_inflate.c
+    arch/s390/s390_features.c
+# S390 *.h
+    arch/s390/dfltcc_common.h
+    arch/s390/dfltcc_deflate.h
+    arch/s390/dfltcc_detail.h
+    arch/s390/dfltcc_inflate.h
+    arch/s390/s390_features.h
+# X86 *.c
+    arch/x86/adler32_avx2.c
+    arch/x86/adler32_avx512_vnni.c
+    arch/x86/adler32_avx512.c
+    arch/x86/adler32_sse42.c
+    arch/x86/adler32_ssse3.c
+    arch/x86/chunkset_avx2.c
+    arch/x86/chunkset_sse2.c
+    arch/x86/chunkset_ssse3.c
+    arch/x86/compare256_avx2.c
+    arch/x86/compare256_sse2.c
+    arch/x86/crc32_pclmulqdq.c
+    arch/x86/crc32_vpclmulqdq.c
+    arch/x86/insert_string_sse42.c
+    arch/x86/slide_hash_avx2.c
+    arch/x86/slide_hash_sse2.c
+    arch/x86/x86_features.c
+# X86 *.h
+    arch/x86/adler32_avx2_p.h
+    arch/x86/adler32_avx512_p.h
+    arch/x86/adler32_ssse3_p.h
+    arch/x86/crc32_fold_pclmulqdq_tpl.h
+    arch/x86/crc32_fold_vpclmulqdq_tpl.h
+    arch/x86/crc32_pclmulqdq_tpl.h
+    arch/x86/x86_features.h
+    arch/x86/x86_intrins.h
+# Generic *.h
+    arch/generic/chunk_permute_table.h
+)
+
+if(HOST_WIN32 OR CLR_CMAKE_TARGET_WIN32)
+    set(ZLIBNG_SOURCES_BASE ${ZLIBNG_SOURCES_BASE} ../../libs/System.IO.Compression.Native/zlibng_allocator_win.c)
+else()
+    set(ZLIBNG_SOURCES_BASE ${ZLIBNG_SOURCES_BASE} ../../libs/System.IO.Compression.Native/zlibng_allocator_unix.c)
+endif()
+
+addprefix(ZLIBNG_SOURCES "${CMAKE_CURRENT_LIST_DIR}/zlib-ng"  "${ZLIBNG_SOURCES_BASE}")
+
+# enable custom zlib allocator
+set(ZLIBNG_COMPILE_DEFINITIONS "MY_ZCALLOC") # TODO: who's using this?
+
+# Compile for zlib-compatible APIs instead of zlib-ng APIs
+set(ZLIBNG_COMPILE_OPTIONS "/zlib-compat")
+
+if(HOST_WIN32 OR CLR_CMAKE_TARGET_WIN32)
+    set(ZLIBNG_COMPILE_OPTIONS "${ZLIBNG_COMPILE_OPTIONS};/wd4127;/wd4131")
+endif()
+
+# TODO: Do i need to pass other arguments or import this directory in a different way?
+# also, are the files in this subdirectory getting correctly modified with the necessary
+# zlibng-compile-definitions and zlibng-compile-options?
+add_subdirectory(zlib-ng)
```

Comment:
@directhex Here's my main question: the zlib-ng.cmake (before called zlib.cmake) is the file that should be doing the preliminary configuration before calling into the proper zlib-ng/CMakeLists.txt.

What I understand this file should do is:
1. Create a variable called ZLIBNG_SOURCES_BASE, which will be a list of all the files from zlib-ng/.
2. Append zlibng_allocator_unix.c or zlibng_allocator_win.c, depending on the platform.
3. Use a custom function created by us (lives under eng/) called addprefix to create the variable ZLIBNG_SOURCES (this one is used in several places) that contains the list of all the files from the _BASE variable but with all the paths prefixed with the full path.
4. Set a definiition MY_ZCALLOC which I have no idea what it is for.
5. Add compilation options: zlib-compat to ensure zlib-ng uses the zlib compatible APIs, and two other options I preserved from zlib: wd4127 (conditional expression is constant) and wd4131 ( uses old-style declarator). I could try deleting the last two, maybe they're not needed anymore in zlib-ng.
6. Add the zlib-ng subdirectory. Interestingly, I couldn't find an equivalent call before this change. So I am unsure how we ensured we reused the official zlib and zlib-intel CMakeLists.txt or .cmake scripts.